packages/app/src/components/ContextSidePanel.tsx

```diff
@@ -55,6 +85,33 @@ export default function ContextSubpanel({
   const formWhere = watch('where');
   const [debouncedWhere] = useDebouncedValue(formWhere, 1000);
 
+  // State management for nested panels
+  const isNested = breadcrumbPath.length > 0;
+
+  const {
+    contextRowId,
+    contextRowSource,
+    setContextRowId,
+    setContextRowSource,
+  } = useNestedPanelState(isNested);
+
+  const { data: contextRowSidePanelSource } = useSource({
+    id: contextRowSource || '',
+  });
+
+  const handleContextSidePanelClose = useCallback(() => {
+    setContextRowId(null);
+    setContextRowSource(null);
+  }, [setContextRowId, setContextRowSource]);
+
+  const handleRowExpandClick = useCallback(
+    (rowWhere: string) => {
+      setContextRowId(rowWhere);
+      setContextRowSource(source.id);
+    },
+    [source.id, setContextRowId, setContextRowSource],
+  );
+
```

Comment:
TL;DR: URL state is used for the side panel. This allows us to expand into local state for nested panels as to not add additional complexity into url state